galahq/gala
# Heroku maintenance, upgrade to Postgres 16 and Redis 7.2 w/ TLS support

The Postgres and Redis instances are reaching EOL support. Heroku is requiring the these data store addons are running the newer versions.

> Heroku Postgres is being deprecated on October 29, 2024 - [heroku.com/changelog-items/2778](https://devcenter.heroku.com/changelog-items/2778).

Changes:
- followed Heroku recommendation to create a new migration that alters the case_search_index materialized view to be compatible with Postgres 16
- upgraded Redis and Sidekiq gems to work with the new TLS requirements for the Heroku Redis upgrade
- Updated docker-compose.yml file from Postgres 13->16
- added the `rake test:unit` to run only the unit tests


To test:
1. prune docker, (`docker compose down`, `docker system prune`, `docker volume prune`, `docker volume rm gala_db_data`)
1. start app with `docker compose up --build`
1. ensure app builds and can be accessed at `http://localhost:3000`, you should see seed data
1. run `docker compose run web rake indices:refresh` inside container to ensure it works
1. trying searching for a case and ensure you are able to see results
1. inspect docker console, ensure you see no Postgres or Redis errors while performing exploratory testing in the app
1. ensure unit tests still pass, `docker compose run web rake test:unit`
